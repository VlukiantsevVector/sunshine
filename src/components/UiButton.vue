<template>
  <button :class="buttonClass" :disabled="inactive" v-on="$listeners">
    <span :class="contentClass" tabindex="-1">
      <slot name="left" />
      <span style="flex: 1">
        <slot />
      </span>
      <slot name="right" />
    </span>
  </button>
</template>

<script lang="ts">
import Vue from 'vue';
import { classHelper } from '@/lib/utils';

const buttonClassHelper = classHelper('ui-btn');
const contentClassHelper = classHelper('ui-btn', 'content');

export default Vue.extend({
  name: 'ui-button',
  inheritAttrs: true,

  props: {
    small: {
      type: Boolean,
      default: false,
    },
    primary: {
      type: Boolean,
      default: false,
    },
    inactive: {
      type: Boolean,
      default: false,
    },
  },

  computed: {
    buttonClass(): object {
      return buttonClassHelper({
        small: this.small,
      });
    },

    contentClass(): object {
      return contentClassHelper({
        main: this.primary,
        secondary: !this.primary,
        inactive: this.inactive,
      });
    },
  },
});
</script>
