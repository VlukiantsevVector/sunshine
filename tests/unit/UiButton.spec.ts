import { mount } from '@vue/test-utils';
import UiButton from '@/components/UiButton.vue';

describe('UiButton', () => {
  it('renders normal button', () => {
    const wrapper = mount(UiButton, {
      slots: {
        default: 'Normal button',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders primary button', () => {
    const wrapper = mount(UiButton, {
      propsData: {
        primary: true,
      },
      slots: {
        default: 'Primary button',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders small button', () => {
    const wrapper = mount(UiButton, {
      propsData: {
        small: true,
      },
      slots: {
        default: 'Small button',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
