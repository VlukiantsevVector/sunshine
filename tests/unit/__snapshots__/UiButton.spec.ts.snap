// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiButton renders normal button 1`] = `<button class="s-btn"><span tabindex="-1" class="inline-flex s-btn__content s-btn__content--secondary"> <span class="flex-grow">Normal button</span> </span></button>`;

exports[`UiButton renders primary button 1`] = `<button class="s-btn"><span tabindex="-1" class="inline-flex s-btn__content s-btn__content--main"> <span class="flex-grow">Primary button</span> </span></button>`;

exports[`UiButton renders small button 1`] = `<button class="s-btn s-btn--small"><span tabindex="-1" class="inline-flex s-btn__content s-btn__content--secondary"> <span class="flex-grow">Small button</span> </span></button>`;
